src/legacy/core_plugins/visualizations/public/np_ready/public/wizard/new_vis_modal.tsx

```diff
@@ -145,8 +145,8 @@ class NewVisModal extends React.Component<TypeSelectionProps, TypeSelectionState
       params = this.props.addBasePath(visType.aliasUrl);
       if (this.props.editorParams && this.props.editorParams.includes('addToDashboard')) {
         params = `${params}?addToDashboard`;
-        this.props.onClose();
```

Comment:
I am not sure, why this `onClose` was called specifically for aliasUrl with `addToDashboard`, but not just `aliasUrl`. This was the reason why for other vis types closing worked, but not for Lens

Reply:
Could it be related to the modal having to stay open for OSS visualizations to pick the index pattern?

Reply:
If I understand you correctly, then this seems like handled separately: 
https://github.com/elastic/kibana/blob/3c429b1ff2b51232646582cb826e709bb2706ed1/src/legacy/core_plugins/visualizations/public/np_ready/public/wizard/new_vis_modal.tsx#L123

And then after search source is selected, code goes to that redirect function.